qml/js/checkSubtitles.js

```diff
@@ -0,0 +1,32 @@
+/**
+
+  Heavily inspired and/or copied from harbour-videoPlayer: https://github.com/llelectronics/videoPlayer
+  original source file: https://github.com/llelectronics/videoPlayer/blob/master/qml/pages/helper/checksubtitles.js
+
+**/
+
+WorkerScript.onMessage = function(message) {
+    var sub = [];
+    var text = [];
+    var a = message.subtitles;
+    var pos = message.position;
+    var ii = 0;
+    var i0 = 0;
+    var i1 = a.length;
+
+    while (i1 - i0 > 1) {
+        ii = (i0 + i1) >> 1;
+        if (pos < a[ii].start) i1 = ii; else i0 = ii;
+    }
+
+    while (i0 >= 0 && pos <= a[i0].end) i0--;
```

Comment:
Maybe this condition could also check for the validity of a[i0] so "i0 >= 0 && a[i0] && pos <= a[i0].end" might be better? I occasionally get these warnings `[W] unknown:22 - file:///usr/share/harbour-nayttamo/qml/js/checkSubtitles.js:22: TypeError: Cannot read property 'end' of undefined`.

Reply:
good catch! I have just been happily using subtitles once they started working and did not spend time on this... :D